### PR TITLE
hack/build-rpms.sh: Update for new builder image

### PR DESCRIPTION
**Pull request by bitoku**
#### What type of PR is this?

/kind ci

#### What this PR does / why we need it:

This PR fixes release-1.29-* CI failure.
I opened the PR https://github.com/openshift/release/pull/54846 to update the build-root image to the rhel9-based one.
The build script also has to be modified accordingly.
Since I use the image from the same repository with different version as the main one, I cherry-picked the commit in the main.

cf. pj-rehearse failure in https://github.com/openshift/release/pull/54846:
https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/54846/rehearse-54846-pull-ci-cri-o-cri-o-release-1.29-images/1816633789661581312

#### Which issue(s) this PR fixes:

This PR and https://github.com/openshift/release/pull/54846 will fix /#8414

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by bitoku** [comment]
cc @haircommander

**Comment by bitoku** [comment]
@cri-o/cri-o-maintainers Can you PTAL?

**Review by saschagrunert — APPROVED** [review]

**Comment by saschagrunert** [comment]
/lgtm cancel

```
hack/build-rpms.sh: line 39: dnf: command not found
```

**Comment by bitoku** [comment]
Should I merge the openshift/release PR first?
https://github.com/openshift/release/pull/54846

**Comment by saschagrunert** [comment]
@bitoku yes having green CI here would be great in the first place before merging.

**Comment by bitoku** [comment]
hmm these PRs are interdependent. so if we merge the CI image change, it won't be green.

**Comment by saschagrunert** [comment]
> hmm these PRs are interdependent. so if we merge the CI image change, it won't be green. To make it green, these 2 PRs should be merged.

If we merge the CI changes, why would this PR not be green? Am I missing something?

**Comment by bitoku** [comment]
sorry for confusing.
yes, it will be green, but the CI change doesn't pass pj-rehearse until this PR gets merged.
Is it ok to merge the CI change first though it fails pj-rehearse?

**Comment by saschagrunert** [comment]
Ah, okay so that might be tricky. We may wanna merge this one first because we have more privileges over this repo then.

/override ci/prow/e2e-aws-ovn
/override ci/prow/e2e-gcp-ovn
/override ci/prow/images

**Pull request merged**
